esp/services/esdl_svc_engine/esdl_store.hpp

```diff
@@ -0,0 +1,82 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _ESDL_STORE_HPP__
+#define _ESDL_STORE_HPP__
+
+#include "jlib.hpp"
+#include "jstring.hpp"
+#include "jptree.hpp"
+
+#ifdef ESDL_ENGINE_EXPORTS
+ #define esdl_engine_decl DECL_EXPORT
+#else
+ #define esdl_engine_decl
+#endif
+
+interface IEsdlStore : public IInterface
+{
+    virtual void fetchDefinition(const char* id, StringBuffer& esxdl) = 0;
+    virtual void fetchDefinition(const char* esdlServiceName, unsigned targetVersion, StringBuffer& esxdl) = 0;
+    virtual void fetchDefinitionByNameOnly(const char * name, StringBuffer & def) = 0;
+    virtual IPropertyTree* fetchBinding(const char* espProcess, const char* bindingName) = 0;
```

Comment:
This whole binding term gets pretty confusing because we have ESP/ENV_Configured/static bindings, and ESDL/DALI_ESDL_Configured/Virtual bindings.  Wish I'd called the ESDL one something else.  Anyway perhaps for this method.. fetchESDLBinding(const char *espProcess, const char *espStaticBinding);  Just to be ultra clear.

Some of your other issues may change this even further because we'll really tie ESDLBindings to a port, rather than a preconfigured ESPStaticBinding.  Eventually maybe we'll get the naming down so it's not so confusing.

Reply:
Changed argument name to espStaticBinding